### ISSUE TITLE
Ksql streams created using CSAS statements create topic automatically with some default schema, is there any way to override this ?

### ISSUE THREAD

**Issue by Worm4047** [confluentinc/ksql#1972]

**Comment by miguno** [comment]
Yes, via the `WITH` clause.  See https://docs.confluent.io/current/ksql/docs/developer-guide/syntax-reference.html#create-table-reserved.

**Comment by Worm4047** [comment]
Yes I tried that, but even after using WITH clause, a new schema is created and takes over the previous schema. The new schema is from ksql data source.

**Comment by miguno** [comment]
Sorry, I meant: What is "the schema" for you?  KSQL's internal data schema?  An Avro schema?  Or do you mean sth else?

**Comment by Worm4047** [comment]
My topic has avro schema.
And one of the field is like :
Id: int
But when CSAS is used the schema is changed to this form:
Id: {int,null}

**Comment by TobiasHauck** [comment]
That's exactly what I'm experiencing right now.

Scenario: You've got two topics that are populated by kafka REST and they've got the same schema. Then you've got a stream streaming the data from one topic to the other based on avro.

The data coming from kafka REST look like ```{"key": "value"}```, the data coming from the stream looks like ```{"key": {"string":"value"}}```. The reason is: KSQL transforms all fields into nullable fields which means the data structure changes.

Is there a way to configure KSQL not to treat everything as nullable, but to use exactly the schema it relies on?

**Comment by vcrfxia** [comment]
Related, since if KSQL can use an existing schema then we have a workaround for this issue: https://github.com/confluentinc/ksql/issues/2427
